### ISSUE TITLE
Why this package is no more on CRAN?

### ISSUE THREAD

**Issue by ecor** [cboettig/knitcitations#107]
I have used this package to write to handle the citatations in RGENERATEPREC CRAN package vignette  but CRAN write to me that this package is no more on CRAN, why? Do you have to solve some issues? May I help you? 

Thanks
@ecor

**Comment by lcolladotor** [comment]
I think that the issue is with `RefManageR` which is imported by `knitcitations` https://github.com/cboettig/knitcitations/blob/master/DESCRIPTION#L21 but was removed from CRAN https://cran.r-project.org/web/packages/RefManageR/index.html. 

I see at https://github.com/ropensci/RefManageR/issues/79 that `RefManageR` just changed maintainers and there was a hiccup which will be resolved based on https://github.com/ropensci/RefManageR/issues/79#issuecomment-718599356

**Comment by ecor** [comment]
Thanks for the reply.
 I found out today that RefManagaR is back on CRAN now (https://github.com/mitchelloharawild/vitae/issues/139) https://cran.r-project.org/web/packages/RefManageR/index.html

**Comment by cboettig** [comment]
Thanks for the ping, I'll take a look at preparing our re-release now

**Comment by ChristopherEeles** [comment]
Hi @cboettig,

I am the maintainer of the PharmacoGx R package. We use knitcitations for our vignettes. Due to the removal of knitcitations form CRAN, our package is failing checks on Bioconductor.

I wondered if you had a time frame for the re-release? I am trying to decide if it is worth removing the dependency or if I can wait for the re-release.

Best,
Christopher Eeles
Software Developer
Benjamin Haibe-Kains Lab
Princess Margaret Cancer Centre

**Comment by cboettig** [comment]
yes, apologies.  I've tried a re-release but CRAN is unhappy about even a soft dependency on bibtex package.  That dependency is needed by RefManageR, which already has the soft dependency and is back on CRAN, but apparently they are hesitant to grant me the same, (I don't really know what's going on, CRAN hasn't been all that responsive about the issue).

My instinct is that if you can switch your vignettes to use RefManageR directly, that might be the way to go.  It's not that different than knitcitations in most of its functionality.

**Comment by ChristopherEeles** [comment]
Thanks for the quick reply. I have also had issues with CRAN being slow to respond in the past. We are a computational biology lab so now we only submit package to Bioconductor. I will look into using RefManageR for our vignettes until you can get things sorted out.

Best,
Chris

**Comment by lcolladotor** [comment]
Hi @cboettig,

I'm also a user of `knitcitations`  and like Chris, I use it for several Bioconductor packages. I appreciate the effort you've put in to resolving the current issue with CRAN. If you think that us as users could write a letter or something to CRAN to help you in your efforts to convince them, please us know and I'd be happy to write such a message.

Best,
Leo

## Thinking out loud


After your reply to Chris, I did look at the code I use from `knitcitations` and while yes, `RefManageR` does a lot, I still use several `knitcitations` functions:

* `citep()` which could be replaced with `RefManageR::Citep()`, but `citep()` also uses `knit_cite()` and in turn `bib_metadata()` + `record_as_cited()` and other internals. I haven't fully grasped whether my recurrent use case benefits from running `knit_cite()` or not.
* `write.bibtex()` which could be replaced with `RefManageR::WriteBib()`
* `bibliography()` which could be replaced with `RefManageR::PrintBibliography()`

Hm.... it seems doable. Though if possible, I'd still like to avoid having to update many R packages :P

**Comment by cboettig** [comment]
just a little random history apropos of nothing really...

I originally wrote `knitcitations` before we had rmardown w/ pandoc-citeproc integration or RefManageR, and really aimed at the online context of generating webpages with embedded semantic citations, e.g. using David Shotton's CITO ontology (see https://www.carlboettiger.info/2013/02/22/semantic-citations-for-the-notebook-and-knitr.html) Given the importance of citations in academic credit, I still think it would have been brilliant to live in a world where we annotated the reason for the citation (e.g. this paper`cito:refutes` claims made by xx, or cito:parodies work of yy. take that, h-index!)  But we don't, and I doubt we'll ever see adoption of such author-provided machine-readable reasons for citations...  

Meanwhile, later tools, particularly first-class pandoc integration into knitr, along with RefManageR's more powerful implementation of DOI metadata extraction (and pubmed ids) means that today I generally find it easier to just conform and rely on `.bib` files and pandoc-style citations.  To keep knitcitations from breaking I've tried to transparently wrap RefManageR in place of what knitcitations used to do by itself rather less robustly.   I think the only thing knitcitations dos that isn't in RefManageR (other than the option to embed CITO semantics mentioned above which I bet no one uses) is the greycite-based support of parsing arbitrary URL metadata (e.g. from dublin core or other `meta` tags).  

Anyway, I'll ping CRAN on my submission again and see if I can get a rationale.  I do worry that if the orphaned `bibtex` package really does bite the dust in future, it will break any useful RefManageR functionality unless @mwmclean  can work around that entirely though(?)

**Comment by jdblischak** [comment]
Because knitcitations was archived, it was automatically removed from the [Reproducible Research CRAN Task View](https://cran.r-project.org/view=ReproducibleResearch). If you're able to get it back on CRAN, please open an Issue at https://github.com/jdblischak/reproducible-research-ctv/issues to let us know to add it back. Thanks!

**Comment by cboettig** [comment]
Hi folks, apologies for the delay, `knitcitations` is back on CRAN now.

Please note, you will have to install `bibtex` package explicitly for any functions which use RefManageR functionality to work (even though RefManageR does not declare that as a hard dependency).  CRAN policy prevents me from including `bibtex` as even a suggested dependency due to it's orphaned status.

**Comment by ChristopherEeles** [comment]
Just wanted to say thanks for putting in the work with the CRAN resubmission. We appreciate it!
